browser/views.py

```diff
@@ -635,10 +635,18 @@ def add_investigator_record(request, person_id):
     if request.method == 'POST':
         form = InvestigatorForm(request.POST, instance=person)
         if form.is_valid():
+            if form.cleaned_data.get('institution_search'):
+                # Create a new institution if it does not exist in the database
+                if not Institution.objects.filter(name=form.cleaned_data.get('institution_search')).exists():
+                    institution = Institution.objects.get_or_create(
+                        name=form.cleaned_data.get('institution_search'),
+                        changed_by=request.user
+                    )
```

Comment:
this is a very unsecure way of doing things. you need to use the primary key, not the name. if two institutions have the same name for some reason, this will lead to problems.

Reply:
Getting institution by id instead of name